pkg/numatopo/interface.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2021 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package numatopo
+
+import (
+	"volcano.sh/apis/pkg/apis/nodeinfo/v1alpha1"
+
+	"volcano.sh/resource-exporter/pkg/args"
+)
+
+// NumaInfo is the interface of resource topolgy data
+type NumaInfo interface {
+	Name() string
+	Update(opt *args.Argument) NumaInfo
+	GetResourceInfoMap() v1alpha1.ResourceInfo
+	GetCPUDetail() map[string]v1alpha1.CPUInfo
```

Comment:
l think this func should be more general

Reply:
use GetResTopoDetail() interface{} to replace it